### PR TITLE
Closes #4761: Integrate feature-search component.

### LINKED ISSUE #4739: Search engines can't be removed
## Steps to reproduce
1. Go to Settings>Search>SearchEngine;
2. In 'Installed search engines' dialog, click more options button and select Remove.
3. Select search engines to remove. -> click "Remove".

### Expected behavior
The search engine selected is removed from the list.

### Actual behavior
Selected search engines can't be removed.

### Device information
* Android device: Samsung Note 10 (Android 11), Google Pixel (Android 10)
* Focus version: dev build from master - #1186.0.1- 20210310152336

### LINKED ISSUE #4761: Replace custom search code with feature-search component
(This needs further investigation / prototyping to break down further)

### PR DESCRIPTION
**Pull request by pocmo**
This patch replaces the custom search code in Focus with our `feature-search` component.

Closes #4739
Closes #4761

### CODE REVIEW

**Comment by pocmo** [comment]
For some reason detekt is crashing with a `ArrayIndexOutOfBoundsException`. I am seeing the same on my "app state prototype" branch. I'll try to update detekt on a separate branch and hopefully that will fix it.

File: app/src/main/java/org/mozilla/focus/search/SearchEngineWriter.kt [hunk 1]
```diff
@@ -1,75 +0,0 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (jonalmeida) [inline app/src/main/java/org/mozilla/focus/search/SearchEngineWriter.kt:1]: Needs test to be deleted for CI to pass.

**Review by jonalmeida — COMMENTED** [review]
The changes look good to me, but I ran into an issue.

While adding a new search engine it would fail with `SocketTimeoutException` when doing a network test of the search engine [here](https://github.com/mozilla-mobile/focus-android/blob/master/app/src/main/java/org/mozilla/focus/settings/ManualAddSearchEngineSettingsFragment.kt#L207).

This seemed related to the changes in https://github.com/mozilla-mobile/focus-android/pull/4717, however when I tried the same URL on a master build, it works fine and I don't know why. 🤔 

#### Steps to reproduce
1. Add a new search engine `https://www.google.com/search?q=%s`.
2. Click 'Save'.

**Comment by pocmo** [comment]
I get an "unknown" error from GV in that situation. After that I also see a background crash and I can't browse websites. It works if I browse a page before going to settings. So I assume that in this situation GV gets initialized wrong somehow. I'll continue to investigate.

**Comment by pocmo** [comment]
> So I assume that in this situation GV gets initialized wrong somehow. I'll continue to investigate.

This doesn't always work either. Interestingly `https://example.org/?q=%s` always works for me. So I wonder if this is somehow Google specific?

**Comment by pocmo** [comment]
That seems to be the case... if I fake the `User-Agent` then it works...

**Comment by pocmo** [comment]
@jonalmeida Can you take another look? Since it's only Google I think we can move this to a new issue and since Google is already in the list, I am thinking it has a super low priority probably.

**Review by jonalmeida — APPROVED** [review]
Looks great! All my custom search engines were migrated too. 🎉

**Pull request merged**